bigtable/admin/instance_admin_client.cc

```diff
@@ -0,0 +1,72 @@
+// Copyright 2018 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "bigtable/admin/instance_admin_client.h"
+#include "bigtable/client/internal/common_client.h"
+
+namespace {
+/**
+ * An AdminClient for single-threaded programs that refreshes credentials on all
+ * gRPC errors.
+ *
+ * This class should not be used by multiple threads, it makes no attempt to
+ * protect its critical sections.  While it is rare that the admin interface
+ * will be used by multiple threads, we should use the same approach here and in
+ * the regular client to support multi-threaded programs.
+ *
+ * The class also aggressively reconnects on any gRPC errors. A future version
+ * should only reconnect on those errors that indicate the credentials or
+ * connections need refreshing.
+ */
+class DefaultInstanceAdminClient : public bigtable::InstanceAdminClient {
```

Comment:
Does this class need to be copyable?

Reply:
Please make it non copyable for now in that case.

Reply:
Done.